src/librbd/internal.cc

```diff
@@ -1176,13 +1141,24 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
   }
 
   int create(IoCtx& io_ctx, const char *imgname, uint64_t size,
-	     ImageOptions& opts,
+             ImageOptions& opts,
+             const std::string &non_primary_global_image_id,
+             const std::string &primary_mirror_uuid) {
+    std::string image_id = util::generate_image_id(io_ctx);
```

Comment:
Minor: can we just remove this version of ```create``` and have its current caller invoke the new version that takes the generated image id?